src/librustc/hir/intravisit.rs

```diff
@@ -607,9 +607,8 @@ pub fn walk_ty<'v, V: Visitor<'v>>(visitor: &mut V, typ: &'v Ty) {
             }
             visitor.visit_lifetime(lifetime);
         }
-        TyImplTraitExistential(item_id, def_id, ref lifetimes) => {
+        TyImplTraitExistential(_, def_id, ref lifetimes) => {
             visitor.visit_def_mention(Def::Existential(def_id));
-            visitor.visit_nested_item(item_id);
```

Comment:
Can you leave a comment here explaining that we don't visit the `impl Trait` ID here because it's collected by `lower_item_id` for the function itself? ([here](https://github.com/rust-lang/rust/pull/51806/files#diff-64b696b0ef6ad44140e973801ed82b25R3129))